src/serializer/serializer.js

```diff
@@ -1498,6 +1580,42 @@ export class Serializer {
       }
     }
 
+    // Inject initializer code for indexed vars into functions
+    for (let [functionValue, funcNode] of funcNodes) {
+      let scopeInitializer = this.functionInitializers.get(functionValue);
+      if (scopeInitializer !== undefined) {
+        let scopeInitializerBody = scopeInitializer.body;
+        if (scopeInitializerBody.length === 0) {
+          this.functionInitializers.delete(functionValue);
+          continue;
+        }
+        let location;
+        for (let value of scopeInitializer.values) {
+          location = this.refs.get(value);
+          if (location !== undefined) break;
+        }
+        if (location === undefined) {
+          // Take care of side-effecting code that is associated with an inlined value
+          location = this.allocateIndexedVar();
+          scopeInitializerBody.unshift(
```

Comment:
No deep reason. Makes generated code easier to read as check and effect are closer together.